SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/RemotePossessionComponent.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/Components/RemotePossessionComponent.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+
+#include "Engine/World.h"
+
+DEFINE_LOG_CATEGORY(LogRemotePossessionComponent);
+
+URemotePossessionComponent::URemotePossessionComponent(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+	, PendingDestroy(false)
+{
+	PrimaryComponentTick.bCanEverTick = true;
+	PrimaryComponentTick.bStartWithTickEnabled = true;
+
+#if ENGINE_MINOR_VERSION <= 23
+	bReplicates = true;
+#else
+	SetIsReplicatedByDefault(true);
+#endif
+}
+
+void URemotePossessionComponent::OnAuthorityGained()
+{
+	AController* Controller = Cast<AController>(GetOwner());
+	ensure(Controller);
+	if (Target == nullptr)
+	{
+		UE_LOG(LogRemotePossessionComponent, Error, TEXT("Target is null"));
```

Comment:
I think we should have a callback for this case, so the user can recover from a case where we tried to possess a pawn that just disappeared.